ambassador/ambassador/mapping.py

```diff
@@ -122,6 +122,53 @@ def get(self, key, *args):
         else:
             return self.attrs.get(key)
 
+    def save_cors_element(self, cors_key, route_key, route):
+        """If self.get('cors')[cors_key] exists, and
+        - is a list, e.g. ["1","2","3"], then route[route_key] is set as "1, 2, 3"
+        - is something else, then set route[route_key] as it is
+
+        :param cors_key: key to exist in self.get('cors'), i.e. Ambassador's config
+        :param route_key: key to save to in envoy's cors configuration
+        :param route: envoy's cors configuration
+        """
+        cors = self.get('cors')
+        if cors.get(cors_key) is not None:
+            if type(cors.get(cors_key)) is list:
+                route[route_key] = ", ".join(cors.get(cors_key))
+            else:
+                route[route_key] = cors.get(cors_key)
+
+    def generate_route_cors(self):
+        """Generates envoy's cors configuration from ambassador's cors configuration
+
+        :return generated envoy cors configuration
+        :rtype: dict
+        """
+
+        cors = self.get('cors')
+        if cors is None:
+            return
+
+        route_cors = {'enabled': True}
+        # cors['origins'] cannot be treated like other keys, because if it's a
+        # list, then it remains as is, but if it's a string, then it's
+        # converted to a list
+        if cors.get('origins') is not None:
```

Comment:
This block will be way simplified by using a temporary variable for `cors.get('origins')` -- it won't make a difference in terms of speed, of course, but it'll make a difference in terms of ease of understanding for the next guy.

Reply:
fixed